src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -110,19 +113,21 @@ public ConnectionProxy(ConnectionUrl connectionUrl, JdbcConnection connection) t
   public static JdbcConnection autodetectClusterAndCreateProxyInstance(ConnectionUrl connectionUrl)
       throws SQLException {
 
-    final IConnectionProvider connectionProvider = new BasicConnectionProvider();
-    final ConnectionProxy connProxy = new ConnectionProxy(connectionUrl, connectionProvider.connect(connectionUrl.getMainHost()));
-    if (connProxy.isPluginEnabled()) {
+    boolean pluginsEnabled = Boolean.parseBoolean(
+      connectionUrl.getConnectionArgumentsAsProperties()
+        .getProperty(PropertyKey.useConnectionPlugins.getKeyName(), Boolean.toString(true)));
```

Comment:
```suggestion
    boolean pluginsEnabled = this.connProps.getBooleanProperty(PropertyKey.useConnectionPlugins).getValue();
```

Reply:
It's not possible to use "this" in static method.